File: WebApi/angular/src/app/shared/model/LoanPaymentSummaryVM.ts
import { DictionaryRecordVM } from '@loancalc/shared/model';
import { MonthlyInstallmentDataVM } from './MonthlyInstallmentDataVm';

export interface ILoanPaymentSummaryVM {
    TotalPayment: number;
    TotalInterest: number;
    TotalLoanAmount: number;
    MonthlyPayment: number;
    MonthlyInterest: number;
    NumberOfMonthlyIntallments: number;
    LoanStartDate: Date;
    LoanPayOffDate: Date;
    Currency: DictionaryRecordVM;
    MonthlyInstallmentDataVM: MonthlyInstallmentDataVM[];
}
export class LoanPaymentSummaryVM implements ILoanPaymentSummaryVM {
    TotalPayment: number;
    TotalInterest: number;
    TotalLoanAmount: number;
    MonthlyPayment: number;
    MonthlyInterest: number;
    NumberOfMonthlyIntallments: number;
    LoanStartDate: Date;
    LoanPayOffDate: Date;
    Currency: DictionaryRecordVM;
    MonthlyInstallmentDataVM: MonthlyInstallmentDataVM[];
}

File: WebApi/Models/Loan/LoanPaymentSummaryVM.cs
using System;

namespace WebApi.Models.Loan
{
    public class LoanPaymentSummaryVM
    {
        public decimal TotalPayment { get; set; }
        public decimal TotalInterest { get; set; }
        public decimal TotalValueOfLoan { get; set; }
        public decimal MonthlyPayment { get; set; }
        public decimal MonthlyInterest { get; set; }
        public int NumberOfMonthlyIntallments { get; set; }
        public DateTime LoanStartDate { get; set; }
        public DateTime LoanPayOffDate { get; set; }
        public DictionaryRecordVM Currency { get; set; }
        public MonthlyInstallmentDataVM[] MonthlyInstallmentData { get; set; }
    }
}